x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/deprecation/DeprecationInfoAction.java

```diff
@@ -78,27 +81,44 @@ private DeprecationInfoAction() {
     }
 
     public static class Response extends ActionResponse implements ToXContentObject {
+        private static final Set<String> RESERVED_NAMES = Sets.newHashSet("cluster_settings", "node_settings");
         private List<DeprecationIssue> clusterSettingsIssues;
         private List<DeprecationIssue> nodeSettingsIssues;
         private Map<String, List<DeprecationIssue>> indexSettingsIssues;
-        private List<DeprecationIssue> mlSettingsIssues;
+        private Map<String, List<DeprecationIssue>> pluginSettingsIssues;
 
         public Response(StreamInput in) throws IOException {
             super(in);
             clusterSettingsIssues = in.readList(DeprecationIssue::new);
             nodeSettingsIssues = in.readList(DeprecationIssue::new);
             indexSettingsIssues = in.readMapOfLists(StreamInput::readString, DeprecationIssue::new);
-            mlSettingsIssues = in.readList(DeprecationIssue::new);
+            if (in.getVersion().before(Version.V_7_11_0)) {
+                List<DeprecationIssue> mlIssues = in.readList(DeprecationIssue::new);
+                pluginSettingsIssues = new HashMap<>();
+                pluginSettingsIssues.put("ml_settings", mlIssues);
+            } else {
+                pluginSettingsIssues = in.readMapOfLists(StreamInput::readString, DeprecationIssue::new);
+            }
         }
 
         public Response(List<DeprecationIssue> clusterSettingsIssues,
                         List<DeprecationIssue> nodeSettingsIssues,
                         Map<String, List<DeprecationIssue>> indexSettingsIssues,
-                        List<DeprecationIssue> mlSettingsIssues) {
+                        Map<String, List<DeprecationIssue>> pluginSettingsIssues) {
             this.clusterSettingsIssues = clusterSettingsIssues;
             this.nodeSettingsIssues = nodeSettingsIssues;
             this.indexSettingsIssues = indexSettingsIssues;
-            this.mlSettingsIssues = mlSettingsIssues;
+            Set<String> reservedNames = new HashSet<>(RESERVED_NAMES);
+            reservedNames.addAll(indexSettingsIssues.keySet());
```

Comment:
Index names are nested under the `index_settings` key, so we don't need to include them in this check.